src/collapse/dtrace.rs

```diff
@@ -155,31 +162,44 @@ impl Folder {
     //     unix`sys_syscall+0x10e
     //       1
     fn on_stack_line(&mut self, line: &str) {
-        let line = line.trim_start();
-        let frame = if self.opt.includeoffset {
-            line
+        let (could_be_java, could_be_cpp, has_colon, mut frame) = if self.opt.includeoffset {
+            (true, true, true, line)
         } else {
             Self::remove_offset(line)
         };
 
-        let mut frame = Self::uncpp(frame);
+        if could_be_cpp {
+            frame = Self::uncpp(frame);
+        }
 
         if frame.is_empty() {
             frame = "-";
         };
 
-        let mut inline = false;
-        for func in frame.split("->") {
-            let mut func = func.trim_start_matches('L').replace(';', ":");
-            if inline {
-                func.push_str("_[i]")
-            };
-            inline = true;
-            self.stack_str_size += func.len() + 1;
-            self.cache_inlines.push(func);
-        }
-        while let Some(func) = self.cache_inlines.pop() {
-            self.stack.push_front(func);
+        if could_be_java {
```

Comment:
I'd probably rename this to `has_inlines`, since it isn't necessarily Java-specific. And also because if it contains `->`, it _does_ have inlines.

Reply:
👍 much better